Warning when build with ESPHome 2025.12.4

I am getting the following:

```
In file included from src/esphome/components/veml6040/veml6040.cpp:1:
src/esphome/components/veml6040/veml6040.h: In member function 'esphome::i2c::ErrorCode esphome::veml6040::VEML6040Component::read_data_register_(uint8_t, uint16_t&)':
src/esphome/components/veml6040/veml6040.h:43:42: warning: 'esphome::i2c::ErrorCode esphome::i2c::I2CDevice::read_register(uint8_t, uint8_t*, size_t, bool)' is deprecated: The stop argument is no longer used. This will be removed from ESPHome 2026.3.0 [-Wdeprecated-declarations]
   43 |         auto retval = this->read_register(a_register, buffer, 2, false);
      |                       ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from src/esphome/components/veml6040/veml6040.h:5:
src/esphome/components/i2c/i2c.h:273:13: note: declared here
  273 |   ErrorCode read_register(uint8_t a_register, uint8_t *data, size_t len, bool stop) {
      |             ^~~~~~~~~~~~~
```

and:

```
In file included from src/esphome.h:127,
                 from src/main.cpp:3:
src/esphome/components/veml6040/veml6040.h: In member function 'esphome::i2c::ErrorCode esphome::veml6040::VEML6040Component::read_data_register_(uint8_t, uint16_t&)':
src/esphome/components/veml6040/veml6040.h:43:42: warning: 'esphome::i2c::ErrorCode esphome::i2c::I2CDevice::read_register(uint8_t, uint8_t*, size_t, bool)' is deprecated: The stop argument is no longer used. This will be removed from ESPHome 2026.3.0 [-Wdeprecated-declarations]
   43 |         auto retval = this->read_register(a_register, buffer, 2, false);
      |                       ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from src/esphome/components/dps310/dps310.h:5,
                 from src/esphome.h:30:
src/esphome/components/i2c/i2c.h:273:13: note: declared here
  273 |   ErrorCode read_register(uint8_t a_register, uint8_t *data, size_t len, bool stop) {
      |             ^~~~~~~~~~~~~
```

for now, it seems to be working fine, but I guess it will break soon (ESPHome 2026.3.0).

Thanks

--- a/components/veml6040/veml6040.h
+++ b/components/veml6040/veml6040.h
@@ -40,7 +40,7 @@ namespace esphome
       i2c::ErrorCode read_data_register_(uint8_t a_register, uint16_t &data)
       {
         uint8_t buffer[2];
-        auto retval = this->read_register(a_register, buffer, 2, false);
+        auto retval = this->read_register(a_register, buffer, 2);
         if (retval == i2c::ERROR_OK) {
           data = (uint16_t(buffer[1]) << 8) | (uint16_t(buffer[0]) & 0xFF);
         }

